File: android/lib/net/request/request.dart
import 'package:android/net/common/message.dart';
import 'package:json_annotation/json_annotation.dart';

part 'request.g.dart';

@JsonSerializable()
class Request extends Message {
  @JsonKey(name: "Action")
  final Action c_action;

  Request(this.c_action) : super(Type.request, UserAgent.client);

  Action get action => c_action;

  factory Request.fromJson(Map<String, dynamic> json) =>
      _$RequestFromJson(json);
  Map<String, dynamic> toJson() => _$RequestToJson(this);
}

enum Action {
  @JsonValue("Action")
  filterStores,
}

File: app/src/main/java/com/github/giorgosathanasopoulos/distributed_systems_aueb/android/backend/network/request/Request.java
package com.github.giorgosathanasopoulos.distributed_systems_aueb.android.backend.network.request;

import com.google.gson.annotations.SerializedName;

import com.github.giorgosathanasopoulos.distributed_systems_aueb.android.backend.json.JsonUtils;
import com.github.giorgosathanasopoulos.distributed_systems_aueb.android.backend.network.Message;

public class Request extends Message {

    @SerializedName("Action")
    private final Action c_Action;

    public Request(Message.UserAgent p_UserAgent, int p_Id, Action p_Action) {
        super(Type.REQUEST, p_Id, p_UserAgent);
        this.c_Action = p_Action;
        setSrc(JsonUtils.toJson(this));
    }

    public Action getAction() {
        return c_Action;
    }

    public enum Action {
        ADD_STORE,
        LIST_STORES, // stats
        FILTER_STORES, // stats

        ADD_PRODUCT,
        LIST_PRODUCTS,
        REMOVE_PRODUCT,
        DECREASE_QUANTITY,
        INCREASE_QUANTITY,
        BUY_PRODUCT,

        SHOW_SALES_FOOD_TYPE, // stats
        SHOW_SALES_STORE_TYPE, // stats

        WORKER_HANDSHAKE, // handshake
        REDUCER_HANDSHAKE, // handshake

        HEARTBEAT,
    }
}
